platform-connectors/main.go

```diff
@@ -154,35 +192,37 @@ func initializeConnectors(
 	config map[string]interface{},
 	stopCh chan struct{},
 	mongoClientCertMountPath string,
-) (*ringbuffer.RingBuffer, *store.MongoDbStoreConnector, error) {
+) (*ringbuffer.RingBuffer, *store.MongoDbStoreConnector, nodemetadata.Processor, error) {
 	var (
 		k8sRingBuffer  *ringbuffer.RingBuffer
 		storeConnector *store.MongoDbStoreConnector
+		processor      nodemetadata.Processor
 		err            error
 	)
 
 	if config["enableK8sPlatformConnector"] == True {
-		k8sRingBuffer, err = initializeK8sConnector(ctx, config, stopCh)
+		k8sRingBuffer, processor, err = initializeK8sConnector(ctx, config, stopCh)
 		if err != nil {
-			return nil, nil, fmt.Errorf("failed to initialize K8s connector: %w", err)
+			return nil, nil, nil, fmt.Errorf("failed to initialize K8s connector: %w", err)
 		}
 	}
 
 	if config["enableMongoDBStorePlatformConnector"] == True {
 		storeConnector, err = initializeMongoDBConnector(ctx, mongoClientCertMountPath)
 		if err != nil {
-			return nil, nil, fmt.Errorf("failed to initialize MongoDB store connector: %w", err)
+			return nil, nil, nil, fmt.Errorf("failed to initialize MongoDB store connector: %w", err)
 		}
 	}
 
-	return k8sRingBuffer, storeConnector, nil
+	return k8sRingBuffer, storeConnector, processor, nil
 }
 
 func cleanupResources(
 	socket string,
 	lis net.Listener,
 	k8sRingBuffer *ringbuffer.RingBuffer,
 	storeConnector *store.MongoDbStoreConnector,
+	processor nodemetadata.Processor,
```

Comment:
I don't think this arg is used in this function, can you please check and remove if it is not used?